### DISCUSSION TITLE
Pyinstaller don't work

### DISCUSSION THREAD

**Discussion by MLangeEU** [community/community#139823] [Programming Help]
### Body

Hi, I'm building a python app with tkinter and now I want to make it an .exe file. 

I have installed pyinstaller
`pip install pyinstaller`

I used the upgrade function
`pip install --upgrade pyinstaller`

But when I try to run a pyinstaller command like `pyinstaller --version` or try to turn the py file into an .exe file
`pyinstaller <filename.py> --onefile`
Powershell responds with a cmdlet error.

Error (translated from German to English, I'm not sure if this is the correct error text)

> pyinstaller : The name “pyinstaller” was not recognized as the name of a cmdlet, function, script file or executable program. Check the spelling of the name, or whether the path is correct (if 
is correct (if included) and repeat the process.

I reinstalled pyinstaller, but it still did not work. Hope someone can help me.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by KhamessiTaha** [comment]
It seems PowerShell isn't recognizing the pyinstaller command because the installation path isn't in your system's PATH environment variable. First, ensure PyInstaller is installed by running pip show pyinstaller in the command prompt. Then, locate the Scripts folder where it's installed (typically under your Python user site-packages directory). Add this folder to your PATH environment variable by following the steps in System Properties > Environment Variables. After updating PATH, restart PowerShell and try running pyinstaller --version again. This should resolve the issue.

**Comment by Giddy-K** [comment]
Hello have you been able to solve the problem?

**Comment by R0b0t-Maker** [comment]
The error likely occurs because pyinstaller isn’t in your system PATH. Here’s how to fix it:

Add PyInstaller to PATH:
   - Locate the Python `Scripts` directory (e.g., `C:\Users\<YourUsername>\AppData\Local\Programs\Python\Python<version>\Scripts`).
   - Add this path to your system's PATH variable through Environment Variables settings.

Restart PowerShell and try running: pyinstaller --version
   
Alternative Command: python -m PyInstaller <filename.py> --onefile

**Comment by YuheshPandian** [comment]
- Try creating a virtual environment and once again install dependencies of the project and run the command in powershell...It worked for me in cmd may be give a try in command prompt.
- If it doesn't work try to add the dependency to the path like how @R0b0t-Maker and @KhamessiTaha  has mentioned. Refer this [blog](https://medium.com/@omoshalewa/why-you-should-add-python-to-path-and-how-58693c17c443) on how to do it.
- Also if it doesn't work at all...try to find an alternative to pyinstaller [like auto-py-to-exe](https://pypi.org/project/auto-py-to-exe/). Refer this [blog](https://medium.com/@asusrishabh/convert-python-program-to-exe-c773345716df) to do so

**Comment by Anuradhasenaratne** [comment]
Response #1

**Comment by Evinrrr** [comment]
First, check if the Python and PyInstaller paths are added to the system environment variables. Second, try running the PyInstaller commands in the command prompt instead of PowerShell. If these don't work, consider creating and using a virtual environment to install and run PyInstaller.
